### PR TITLE
Fix/ Mnemonic verification fails

### LINKED ISSUE #104: Regenerate mnemonic if verification fails
- user chooses wrong nth word when verifying mnemonic, go back and refresh mnemonic
- display error notification about failed mnemonic verification

### PR DESCRIPTION
**Pull request by irubido**
resolves #104

### CODE REVIEW

**Comment by mpetrunic** [comment]
@irubido try to describe in Pr descrition how did you solve the problem :)

File: src/renderer/constants/action-types.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export enum RegisterActionTypes {
+    STORE_FAILED_VERIFICATION = "STORE_FAILED_VERIFICATION",
```
Comment (mpetrunic) [inline src/renderer/constants/action-types.ts]: rename to SET_SIGNING_NEMONIC_VERIFICATION_STATUS

File: src/renderer/reducers/register.ts [hunk 2]
```diff
@@ -1,21 +1,28 @@
...
-    withdrawalKey: ""
+    withdrawalKey: "",
+    failedVerification: false
 };
 
 export const registerReducer = (state = initialState, action: Action<RegisterActionTypes>): IRegisterState => {
     switch (action.type) {
+        case RegisterActionTypes.STORE_FAILED_VERIFICATION:
+            return Object.assign({}, state, {
+                failedVerification: (action as IFailedVerificationAction).payload.failedVerification
```
Comment (mpetrunic) [inline src/renderer/reducers/register.ts]: you should probably put `mnemonic: ""`
and just put `failedSigningMnemonicVerification: true` - no need for payload

File: src/renderer/actions/register.ts [hunk 3]
```diff
@@ -10,6 +10,23 @@ import {KEYSTORE_DEFAULT_DIRECTORY} from "../constants/keystore";
 import {CGAccount} from "../models/account";
 import {getV4Filename} from "../services/utils/crypto-utils";
 
+// Mnemonic Failed Verification
+export const storeFailedVerificationAction = (failedVerification: boolean) =>
+    (dispatch: Dispatch<IFailedVerificationAction>): void => {
+        dispatch(setFailedVerification(failedVerification));
+    };
+
+export const setFailedVerification = (failedVerification: boolean): IFailedVerificationAction => ({
```
Comment (mpetrunic) [inline src/renderer/actions/register.ts]: storeSigningMnemonicVerificationStatus

- rename on other places as well to properly describe action

File: src/renderer/containers/Onboard/SigningKey/Verify/SigningMnemonicQuestion.tsx [hunk 4]
```diff
@@ -1,28 +1,38 @@
...
+            this.props.storeFailedVerification(true);
+            this.props.history.goBack();
+        };
+        
         return (
             <VerifyMnemonic
                 question={`What’s the ${correctAnswerIndex + 1}th word in the mnemonic?`}
                 answers={[mnemonic[randArray[0]], mnemonic[randArray[1]], mnemonic[randArray[2]]]}
                 correctAnswer={mnemonic[correctAnswerIndex]}
                 onCorrectAnswer={(): void => {}}
```
Comment (mpetrunic) [inline src/renderer/containers/Onboard/SigningKey/Verify/SigningMnemonicQuestion.tsx]: I don't think you have latest changes to master, this should navigate somewhere and clear signing mnemonic verification status

**Comment by mpetrunic** [comment]
@irubido It's a bit awkward transition. Can you on invalid answer, disable all other buttons and redirect after like 1s delay?

**Comment by irubido** [comment]
> @irubido It's a bit awkward transition. Can you on invalid answer, disable all other buttons and redirect after like 1s delay?

**Comment by irubido** [comment]
-redirecting with setTimeout after 1 second
-buttons disable after a chosen answer, added disable style for ButtonInverted and ButtonDestructive

**Pull request merged**